src/core/server/saved_objects/migrations/core/document_migrator.ts

```diff
@@ -201,11 +237,19 @@ function buildActiveMigrations(
   typeRegistry: ISavedObjectTypeRegistry,
   log: Logger
 ): ActiveMigrations {
-  return typeRegistry
-    .getAllTypes()
-    .filter((type) => type.migrations && Object.keys(type.migrations).length > 0)
+  const typesWithMigrationMaps = typeRegistry.getAllTypes().map((type) => ({
+    ...type,
+    migrationsMap: typeof type.migrations === 'function' ? type.migrations() : type.migrations,
+  }));
```

Comment:
NIT: I see you did it below, but I think we could filter on `Boolean(type.migrations)` to remove the types that don't specify `migrations`, to avoid calling `validateMigrationsMapObject` for them?